### PR TITLE
fix: image asset ids

### PR DESCRIPTION
**Pull request by samuelOsborne**

### CODE REVIEW

File: packages/dotlottie-js/src/common/dotlottie-common.ts [hunk 1]
```diff
@@ -245,18 +246,33 @@ export class DotLottieCommon {
...
+   * @returns Map of duplicate image ids and their count.
+   */
+  private _getMapOfDuplicateImageIds(): Map<string, number> {
+    let count = 0;
+    const dupeMap = new Map<string, number>();
 
     this.animations.forEach((animation) => {
-      images.set(animation.id, animation.imageAssets);
+      animation.imageAssets.forEach((imageAsset) => {
+        if (dupeMap.has(imageAsset.id)) {
```
Comment (theashraf) [inline packages/dotlottie-js/src/common/dotlottie-common.ts:259]: The `id` is not always the same as the `filename`, for example 

id: `image_0`, filename: `image_0.png`
is not the same as 
id: `image_0`, filename: 'image_0.webp'

right ? probably our ids should be the filename and we can get rid of using the image id in this case, no ?

File: packages/dotlottie-js/src/common/dotlottie-common.ts [hunk 2]
```diff
@@ -245,18 +246,33 @@ export class DotLottieCommon {
...
   }
 
-  private _renameImageAssets(): void {
-    const images: Map<string, LottieImageCommon[]> = new Map();
+  /**
+   * Generates a map of duplicate image ids and their count.
+   * @returns Map of duplicate image ids and their count.
+   */
+  private _getMapOfDuplicateImageIds(): Map<string, number> {
+    let count = 0;
```
Comment (theashraf) [inline packages/dotlottie-js/src/common/dotlottie-common.ts:254]: Why do we need this global `count` variable ? isn't the map already has the occurrence of each image by its ID ?

File: packages/dotlottie-js/src/common/dotlottie-common.ts [hunk 3]
```diff
@@ -245,18 +246,33 @@ export class DotLottieCommon {
...
+    const dupeMap = new Map<string, number>();
 
     this.animations.forEach((animation) => {
-      images.set(animation.id, animation.imageAssets);
+      animation.imageAssets.forEach((imageAsset) => {
+        if (dupeMap.has(imageAsset.id)) {
+          count = dupeMap.get(imageAsset.id) ?? 0;
+          dupeMap.set(imageAsset.id, count + 1);
+        } else {
+          dupeMap.set(imageAsset.id, 1);
```
Comment (theashraf) [inline packages/dotlottie-js/src/common/dotlottie-common.ts:263]: This seems to be an `occuranceMap` not `duplicateMap`, right ? as based on this line, even if there is a unique image id, still would be pushed to the map with an occurance of `1`

File: packages/dotlottie-js/src/common/lottie-image-common.ts [hunk 4]
```diff
@@ -140,6 +140,8 @@ export class LottieImageCommon {
       }
       // Default to png if the file extension isn't available
       this.fileName = `${newName}.${fileExt}`;
+    } else {
+      this.fileName = `${newName}.png`;
```
Comment (theashraf) [inline packages/dotlottie-js/src/common/lottie-image-common.ts:144]: Isn't the extension inferred from the mimetype of the data ? defaulting it to png is safe ?

**Review by theashraf — COMMENTED** [review]
I think we need to establish a clear distinction between the following in the LottieImage class:

- `id`
- `fileName`
In this update, we're relying on fileName as the new identifier, so we might no longer need the id property for images.

or, we could continue using `id` as the filename, inferring the extension automatically from the image data ?

File: packages/dotlottie-js/src/common/dotlottie-common.ts (hunk 1 above)
Comment (samuelOsborne) [reply packages/dotlottie-js/src/common/dotlottie-common.ts:259]: Inside the lottie itself, images had an id field, and then a name and path attribute, thats why I mapped it like this to the image class.

The id is just the filename without the extension
In your example the id stayed the same despite changing the extension

File: packages/dotlottie-js/src/common/dotlottie-common.ts (hunk 3 above)
Comment (samuelOsborne) [reply packages/dotlottie-js/src/common/dotlottie-common.ts:263]: It was duplicateMap as duplicate filename not the content of the image, will change to filename occurence map

**Review by theashraf — COMMENTED** [review]
Let's add a changeset ?

**Review by theashraf — APPROVED** [review]

**Pull request merged**